src/ILVerification/src/VerificationResult.cs

```diff
@@ -16,6 +17,9 @@ public class VerificationResult
     public struct VerificationErrorArgs
     {
         public VerifierError Code { get; internal set; }
+        public int TokenClass { get; internal set; }
+        public int TokenInterface { get; internal set; }
+        public int TokenMethod { get; internal set; }
```

Comment:
@jkotas here I've got some problem. You said that you weren't convinced about `VerificationResult` refactoring. What do you think we need to do here?The main issue is that I need a way to distinguish between method verification result and type verification result to pick up correct token value and format a proper string [here](https://github.com/dotnet/corert/blob/master/src/ILVerify/src/Program.cs#L152)

Reply:
I think it is hard to format a proper string out of a fixed components outside the type verifier. The type verifier errors are going include one or more type names; one or more method names; or one or more field names.

I think this would need to be array of arbitrary arguments. The fixed set of 3 tokens is not going to work well as we add more type verification errors.

Reply:
> arbitrary arguments

Do you mean somthing like
```cs
public class VerificationErrorArgs : Dictionary<string,object>
{
   public VerifierError Code { get; internal set; }
}

or

public class VerificationErrorArgs : List<Arg>
{
   public VerifierError Code { get; internal set; }
}

public struct Arg
{
   public string Name { get; set; }
   public object Value { get; set; }
   public Type Type { get; set; }
}

```
and on Program.cs check `Code` value and print values? We could have string/int in same args list...for instance method args Token/Found. I don't like object for boxing...but it's likely I did't fully understand your idea.

Reply:
Can the argument be just an `object[]` ? You would product the nice message to print by simply running `string.Format` on it.

Reply:
> I don't like object for boxing.

Boxing is perfectly fine for formatting error messages.

Reply:
> Can the arguments be just an object[] ? 

Printing helper on command line does this https://github.com/dotnet/corert/blob/master/src/ILVerify/src/Program.cs#L175 if we return object[] we need to fix indexer value, right? I mean in case of new arguments we could introduce errors.

Reply:
Yes, this is very specific to IL verification errors. This needs generalizing to handle more types of errors.

Reply:
I added an attempt to generalize `VerificationResult`. Idea was to follow current idea of one message assembled by verifier and a console print assembled by driver(I don't know if driver is a correct definition, but on discussions I found this concept->term, in this case utilizator of `ILVerification`).

Reply:
I tried this idea because here https://github.com/dotnet/corert/blob/master/src/ILVerify/src/Program.cs#L178 current code test "arg name" before print...so we need a way to "give a name" to arg. Sometimes some args are missing, with simple `object[]` isn't a way to know what arg is.